File: Célia_LingProgram/src /Lista02/Exer03_SistemaDeDesconto.java
package Lista02;

import java.util.Scanner;

public class Exer03_SistemaDeDesconto {
    public static void main(String[] args) {

        Double valor, desconto = 0.0, valorADesc;
        Integer codigo;

        Scanner leitor = new Scanner(System.in);

        System.out.println("Desconto");
        System.out.println("Insira o valor do produto escolhido");
        valor = leitor.nextDouble();
        System.out.println("Insira o codigo de desconto");
        codigo = leitor.nextInt();

        switch (codigo) {
            case 1:
                desconto = 0.05;
                break;
            case 2:
                desconto = 0.1;
                break;
            case 3:
                desconto = 0.2;
                break;
            case 4:
                desconto = 0.5;
                break;
            default:
                System.out.println("Código inválido");
                break;
        }

        if (desconto > 0) {
            valorADesc = desconto * valor;
            System.out.println("Valor sem desconto: R$"+valor);
            System.out.println("Valor com desconto: R$" + (valor - valorADesc) );
            System.out.println("Voce economiza R$"+ valorADesc);
        } else {
            System.out.println("Solicite o desconto novamente");
        }




    }
}

File: Célia_LingProgram/JavaFacul/src/Lista02/Exer03_SistemaDeDesconto.kt
package Lista02

import java.util.*

//Crie uma classe chamada “Desconto” que:
//A. Solicite o valor do produto escolhido.
//B. Solicite o código de desconto.
//O desconto deve ser calculado de acordo com o código fornecido,
//conforme a tabela
//abaixo:
//Código Desconto
//Desconto
//1 5%
//2 10%
//3 20%
//4 50%
//C. Exiba o valor do produto sem desconto.
//D. Exiba o valor do produto com o desconto.
//E. Exiba uma mensagem de código inválido caso o usuário digite um código
//inexistente. Dica: utilize o switch/case

fun main() {
    val reader = Scanner(System.`in`)

    println("Desconto")
    println("Insira o valor do seu produto: ")
    val valorProd: Double = reader.nextDouble()
    println("Insira o código de desconto: ")
    val codDesc: Int = reader.nextInt()

    when (codDesc) {
        1 -> println("Valor sem desconto: R$" + valorProd + "\n Valor com desconto: R$" + (valorProd - (valorProd * 0.05)))
        2 -> println("Valor sem desconto: R$" + valorProd + "\n Valor com desconto: R$" + (valorProd - (valorProd * 0.10)))
        3 -> println("Valor sem desconto: R$" + valorProd + "\n Valor com desconto: R$" + (valorProd - (valorProd * 0.20)))
        4 -> println("Valor sem desconto: R$" + valorProd + "\n Valor com desconto: R$" + (valorProd - (valorProd * 0.50)))
        else -> println("Código de desconto inválido.")
    }
}